src/components/DaySelect/calander_component/calander_component.tsx

```diff
@@ -88,14 +89,58 @@ export const CalanderComponent = ({
       return a.getTime() - b.getTime();
     });
 
+    setLastSelectedDate(date);
     setSelectedDates(arr);
   };
 
   const removeDay = (date: Date) => {
     const arr = selectedDates.filter((obj) => obj.getTime() != date.getTime());
+    setLastSelectedDate(date);
     setSelectedDates(arr);
   };
 
+  const handleRange = (date: Date) => {
+    if (lastSelectedDate) {
+      const range = getRange(lastSelectedDate, date);
+      let arr = [...selectedDates];
+      range.forEach((day) => {
```

Comment:
pretty slick implementation, but hard to read at a glance -- isolate some of this work to functions. For instance on line 108->112 could be a toggleDateSelection func